src/System.Diagnostics.StackTrace.Symbols/src/System/Diagnostics/StackTrace/Symbols.cs

```diff
@@ -0,0 +1,204 @@
+                        if (!methodInfo.SequencePointsBlob.IsNil)
+                        {
+                            SequencePointCollection sequencePoints = methodInfo.GetSequencePoints();
+
+                            SequencePoint? bestPointSoFar = null;
+                            foreach (SequencePoint point in sequencePoints)
+                            {
+                                if (point.Offset > ilOffset)
+                                    break;
+
+                                if (point.StartLine != SequencePoint.HiddenLine)
+                                    bestPointSoFar = point;
+                            }
+
+                            if (bestPointSoFar.HasValue)
+                            {
+                                sourceLine = bestPointSoFar.Value.StartLine;
+                                sourceColumn = bestPointSoFar.Value.StartColumn;
+                                sourceFile = reader.GetString(reader.GetDocument(bestPointSoFar.Value.Document).Name);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the portable PDB reader for the assembly path
+        /// </summary>
+        /// <param name="assemblyPath">file path of the assembly or null</param>
+        /// <param name="inMemoryAddress">in memory PE (assemblyPath != null) or PDB (assemblyPath == null) address or zero</param>
+        /// <param name="inMemorySize">in memory PE or PDB size</param>
+        /// <param name="reader">returns the reader</param>
+        /// <returns>provider</returns>
+        static IDisposable GetReader(string assemblyPath, IntPtr inMemoryAddress, int inMemorySize, out MetadataReader reader)
+        {
+            reader = null;
+            MetadataReaderProvider provider = null;
+
+            if (assemblyPath != null)
+            {
+                uint stamp;
+                int age;
+                Guid guid;
+
+                string pdbName = GetPdbPathFromPeStream(assemblyPath, inMemoryAddress, inMemorySize, out age, out guid, out stamp);
+                if (pdbName != null)
+                {
+                    try
+                    {
+                        Stream pdbStream = File.OpenRead(pdbName);
+
+                        // Need to always return provider so that it will be disposed
+                        provider = MetadataReaderProvider.FromPortablePdbStream(pdbStream);
+                        MetadataReader rdr = provider.GetMetadataReader();
+
+                        // Validate that the PDB matches the assembly version
+                        if (age == 1)
+                        {
+                            if (IdEquals(rdr.DebugMetadataHeader.Id, guid, stamp))
+                            {
+                                reader = rdr;
+                            }
+                        }
+                    }
+                    catch
```

Comment:
Should we be catching more specific exceptions?  Or we really do want to eat everything?